compressor/Bit_cashier.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include <deque>
+#include <string>
+
+class Bit_cashier
+{
+  std::deque<bool> bits;
+  const int byte_length = 8;
+
+  void put_error_and_exit(const std::string& message)
+  {
+    std::cerr << message;
+    exit(1);
+  }
+
+public:
+  Bit_cashier(){}
+
+  void add(const bool new_value)
+  {
```

Comment:
You can format this much shorter:
void add(const bool new_value)
  {bits.push_back(new_value);}